awesome/src/lua/config.rs

```diff
@@ -43,24 +43,39 @@ pub fn load_config(mut lua: &mut Lua, cmdline_config: Option<&str>, lib_paths: &
             init_file
                 .read_to_string(&mut init_contents)
                 .expect("Could not read contents");
-            lua.exec(init_contents.as_str(), Some("init.lua".into()))
+            lua.load(init_contents.as_str())
+                .set_name("init.lua")
+                .unwrap()
+                .exec()
                 .map(|_: ()| info!("Read init.lua successfully"))
                 .or_else(|err| {
                     log_error(err);
                     info!("Defaulting to pre-compiled init.lua");
                     unsafe {
-                        *lua = Lua::new_with_debug();
+                        // TODO: overwrite global lua!
```

Comment:
I'm guessing the way you will have to achieve this is that instead of passing in a `Context` here for `lua` you should pass in a `Lua`. Then you can replace the lua instance properly.

Note that the reason we replace the whole Lua instance here (in case you didn't already guess) is that this is the case where the init script was unsucessfully executed. However, the previous state from that execution is still hanging around and so we need to clear it out by making an entirely new instance.